app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEModule.java

```diff
@@ -183,7 +184,8 @@ FieldViewModelFactory fieldViewModelFactory(
                 ),
                 new UiEventTypesProviderImpl(),
                 new KeyboardActionProviderImpl(),
-                new LegendValueProviderImpl(d2, resourceManager));
+                new LegendValueProviderImpl(d2, resourceManager),
+                null);
```

Comment:
I wouldn't make it null I will provide it always.

Reply:
Updated